src/main/java/com/recurse/portfolio/data/DisplayAuthor.java

```diff
@@ -15,7 +15,7 @@ public static DisplayAuthor fromUserForUser(
     ) {
         switch(sourceUser.getProfileVisibility()) {
             case PRIVATE:
-                if (sourceUser == requestingUser) {
+                if (sourceUser.equals(requestingUser)) {
```

Comment:
Thank you for catching and fixing this! And thank you for calling it out in a separate commit.